app/controllers/index.js

```diff
@@ -1,30 +1,23 @@
-import { match, not } from '@ember/object/computed';
 import Controller from '@ember/controller';
+import { action } from '@ember/object';
+import { match, not } from '@ember/object/computed';
 
-export default Controller.extend({
-
-  headerMessage: 'Demo Home Page',
-  responseMessage: '',
-  emailAddress: '',
-
-  isValid: match('emailAddress', /^.+@.+\..+$/),
-  isDisabled: not('isValid'),
-
-  actions: {
+export default class HomeController extends Controller {
 
-    saveInvitation() {
-      const email = this.emailAddress;
+  headerMessage = 'Demo Home Page';
+  responseMessage = '';
+  emailAddress = '';
 
-      const newInvitation = this.store.createRecord('invitation', {
-        email: email
-      });
+  @match('emailAddress', /^.+@.+\..+$/) isValid;
+  @not('isValid') isDisabled;
 
-      newInvitation.save().then((response) => {
-        this.set('responseMessage', `Thank you! We saved your email address with the following id: ${response.get('id')}`);
-        this.set('emailAddress', '');
-      });
+  @action
+  saveInvitation() {
+    const newInvitation = this.store.createRecord('invitation', { email: this.emailAddress });
 
-    }
+    newInvitation.save().then((response) => {
```

Comment:
We could destructure the function parameter here, even though I realized recently that readability may suffer.

```js
newInvitation.save().then(({ id }) => { ...following id: ${id} });
```

At the very minimum, let's get rid of the parentheses.

```js
newInvitation.save().then(response => { ... }
```

Reply:
You right, we can remove parenthesis.

This part of the app shows up at the beginning of the tutorial, where it is better to show less magic and advance JS tricks, so I would keep it more simple at the moment.